pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -802,6 +802,67 @@ public class MyTest {
         </example>
     </rule>
 
+    <rule name="JUnit5TestShouldBePackagePrivate"
+          language="java"
+          since="6.35.0"
+          message="JUnit 5 tests should be package-private."
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          typeResolution="true"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#junit5testshouldbepackageprivate">
+        <description>
+JUnit 5 tests should be package private at the class level and for each method that uses @Test, @RepeatedTest,
+@TestFactory, @TestTemplate or @ParameterizedTest.
+Contrary to JUnit4 tests that required public visibility to be run by the engine, JUnit5 tests may be run
```

Comment:
Thanks for updating the description! Minor suggestion:

> JUnit5 tests may be run only with package-private visibility

This sounds like they *require* package-private visibility. IMO a better way to put it would be "JUnit5 tests can also be run if they're package-private"

Reply:
Done